### PR TITLE
Add support for multiple parallel popen calls

### LINKED ISSUE #90: Parallel uses of mock Popen cross-contaminate
If two calls to Popen are made, and the first is accessed after the second occurs, the outputs of the second feed into the first.

### PR DESCRIPTION
**Pull request by palfrey**
Fixes #90

### CODE REVIEW

**Comment by palfrey** [comment]
This appears to be failing for `coveralls-check` issues (which I'm fairly sure has nothing to do with my PR)

```
$ coveralls-check $TRAVIS_COMMIT --parallel-build-number $TRAVIS_BUILD_NUMBER --repo-token $COVERALLS_REPO_TOKEN
usage: coveralls-check [-h] [--fail-under FAIL_UNDER] [--max-wait MAX_WAIT]
                       [--parallel-build-number PARALLEL_BUILD_NUMBER]
                       [--repo-token REPO_TOKEN]
                       commit
coveralls-check: error: argument --repo-token: expected one argument```

**Comment by cjw296** [comment]
Yeah, see https://github.com/travis-ci/travis-ci/issues/10117 and https://github.com/lemurheavy/coveralls-public/issues/1139 for the backstory on the coveralls check failing. It's so frustrating that there isn't a viable alternative.

File: testfixtures/tests/test_popen.py [hunk 1]
```diff
@@ -31,9 +31,9 @@ def test_command_min_args(self):
         compare(process.returncode, 0)
         # test call list
         compare([
-                call.Popen('a command', stderr=-1, stdout=-1),
-                call.Popen_instance.communicate(),
-                ], Popen.mock.method_calls)
+            call.Popen('a command', stderr=-1, stdout=-1),
+        ], Popen.mock.method_calls)
+        compare([call.communicate()], process.method_calls)
```
Comment (cjw296) [inline testfixtures/tests/test_popen.py]: Hmm, no, not a fan of this, mainly as it's massively backwards incompatible.

**Comment by palfrey** [comment]
That gets pretty tricky, as when a call is made to `returncode` or similar, you'd have to know *which* one of the various returns this was related to. Splitting it out into a separate class was the cleanest method I could think of doing this. If you've got any suggestions about how to do this another way, I'd love to hear them.

**Comment by cjw296** [comment]
https://testfixtures.readthedocs.io/en/latest/popen.html#multiple-process-executions

**Comment by palfrey** [comment]
Except that those different behaviours bleed through. If I've got exitcode 1 on the first, and exitcode 2 on the second, after two calls the returncode of both is 2, no matter whether I use callable behaviours or a pair of set_command's.

**Pull request closed without merging**